### PR TITLE
PP-270: Node/queue association is ignored when node_group_key is only…

### PR DESCRIPTION
**Pull request by bhroam**
#### Issue-ID

* PP-270

#### Problem description
Node/queue association is ignored when node_group_key is set at the server level and not the queue level

#### Cause / Analysis
The scheduler will use a queue's placement sets if they are available. The scheduler would only create queue level placement sets if node_group_key was set at the queue. Since no placement sets were on the queue, the server's sets would be used. The server's placement sets do not enforce node/queue association.
#### Solution description
Create placement sets for a queue if the queue has nodes associated with it or the queue has a node_group_key set

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
- [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [x] I have added tests to cover my changes. To create automated tests, see **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**.
- [x] All new and existing automated tests have passed. See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**.
- [x] I have attached automated (PTL)/manual test logs to the associated Jira ticket.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

File: test/tests/pbs_nodes_queues.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        self.server.manager(MGR_CMD_UNSET, NODE, 'queue', id='vnode[3]')
+
+    def test_pbs7100(self):
+        """
+        Issue with node grouping and nodes associated with queues.  If
+        node_grouping is set at the server level, node/queue association is
+        not honored
+        """
+
+       a = {'Resource_List.select': '2:ncpus=1', 'Resource_List.place': 'vscatter', 'queue': 'workq2'}
```
Comment (mike0042) [inline test/tests/pbs_nodes_queues.py]: Indentation is off here. Please ensure the test is PEP8 compliant.

Comment (bhroam) [inline test/tests/pbs_nodes_queues.py]: I don't see the incorrect indentation.  Where is it?  Also, I ran autopep8.

**Comment by bayucan** [comment]
Code fix itself looks good. Your PTL test file is missing the licensing header...

File: test/tests/pbs_nodes_queues.py (hunk 1 above)
Comment (mike0042) [inline test/tests/pbs_nodes_queues.py]: At line 53. It looks like "a =" is preceded by seven spaces rather than eight. Easier to see it in context.

**Comment by mike0042** [comment]
I sign off. You must update your branch now that PP-123 has been merged.

**Comment by subhasisb** [comment]
@bhroam have you attached the test logs to the Jira ticket as evidence of testing?

**Comment by bayucan** [comment]
Looks good. I sign off.

**Comment by bhroam** [comment]
I've added the PTL test output added to PP ticket.  I've squashed my commits.  I believe this pull request is ready for merging.

**Pull request merged**